saga/seata-saga-spring/src/main/java/org/apache/seata/saga/engine/store/db/DbAndReportTcStateLogStore.java

```
@@ -729,11 +729,26 @@ public List<StateInstance> queryStateInstanceListByMachineInstanceId(String stat
     }
 
     @Override
-    public void clearUp() {
+    public void clearUp(StateMachineInstance machineInstance, ProcessContext context) {
         RootContext.unbind();
         RootContext.unbindBranchType();
         if (sagaTransactionalTemplate != null) {
-            sagaTransactionalTemplate.cleanUp();
+            GlobalTransaction globalTransaction;
+            try {
+                globalTransaction = getGlobalTransaction(machineInstance, context);
+                if (globalTransaction == null) {
```

Why not unify it in template?

---

Do you mean to move the getGlobalTransaction method to the template? If so, it should be a refactoring of the code and should not be done in this commit.
你的意思是把getGlobalTransaction 方法移动到模版里么 如果这样的话应该是对代码的一次重构,不应该在本次提交中完成

---

> Do you mean to move the getGlobalTransaction method to the template? If so, it should be a refactoring of the code and should not be done in this commit. 你的意思是把getGlobalTransaction 方法移动到模版里么 如果这样的话应该是对代码的一次重构,不应该在本次提交中完成

我说的是在template中针对传入globaltransactional进行判空,如果是空的应该抛异常提示不允许为null
I'm talking about passing in globaltransactional in the template for null, if it's null it should throw an exception to indicate that it's not allowed to be null.